### PR TITLE
Add simpler and faster debug Makefile target

### PR DESCRIPTION
**Pull request by mpl**
### What does this PR do?

This PR adds new targets in order to build faster (especially on Mac) docker images, for easier iteration while debugging on e.g. Kubernetes. The generated image is also alpine-based, with tools, instead of being from scratch.
It also cleans up a few things to ease maintainability/readability.

### Motivation

To ease the debugging workflow on Kubernetes.

### More

~- [ ] Added/updated tests~
~- [ ] Added/updated documentation~

### CODE REVIEW

**Comment by mpl** [comment]
Blocking it for now because I suspect it broke the "optimization" for running the integration tests faster.

**Review by rtribotte — APPROVED** [review]
LGTM 👌

File: Makefile [hunk 1]
```diff
@@ -125,6 +127,10 @@ build-image: clean-webui binary
 build-image-dirty: binary
 	docker build -t $(TRAEFIK_IMAGE) .
 
+## Locally build traefik for linux, then shove it an ubuntu image, with basic tools.
```
Comment (rtribotte) [inline Makefile:130]:
```suggestion
## Locally build traefik for linux, then shove it an alpine image, with basic tools.
```

**Comment by mpl** [comment]
blocking the merging for now as we need to figure out the apk upgrade question (and be consistent in all Dockerfiles)

**Review by tomMoulard — APPROVED** [review]

**Pull request merged**